src/Aspire.Hosting.AWS/Lambda/LambdaLifecycleHook.cs

```diff
@@ -0,0 +1,126 @@
+        {
+            return;
+        }
+
+        var expectedVersion = emulatorAnnotation.OverrideMinimumInstallVersion ?? DefaultLambdaTestToolVersion;
+        var installedVersion = await GetCurrentInstalledVersionAsync(cancellationToken);
+
+        if (ShouldInstall(installedVersion, expectedVersion, emulatorAnnotation.AllowDowngrade))
+        {
+            logger.LogDebug("Installing .NET Tool Amazon.Lambda.TestTool ({version})", installedVersion);
+
+            var commandLineArgument = $"tool install -g Amazon.Lambda.TestTool --version {expectedVersion}";
+            if (emulatorAnnotation.AllowDowngrade)
+            {
+                commandLineArgument += " --allow-downgrade";
+            }
+
+            var result = await processCommandService.RunProcessAndCaptureOuputAsync(logger, "dotnet", commandLineArgument, cancellationToken);
+            if (result.ExitCode == 0)
+            {
+                if (!string.IsNullOrEmpty(installedVersion))
+                {
+                    logger.LogInformation("Successfully Updated .NET Tool Amazon.Lambda.TestTool from version {installedVersion} to {newVersion}", installedVersion, expectedVersion);
+                }
+                else
+                {
+                    logger.LogInformation("Successfully installed .NET Tool Amazon.Lambda.TestTool ({version})", expectedVersion);
+                }
+            }
+            else
+            {
+                if (!string.IsNullOrEmpty(installedVersion))
+                {
+                    logger.LogWarning("Failed to update Amazon.Lambda.TestTool from {installedVersion} to {expectedVersion}:\n{output}", installedVersion, expectedVersion, result.Output);
+                }
+                else
+                {
+                    logger.LogError("Fail to install Amazon.Lambda.TestTool ({version}) required for running Lambda functions locally:\n{output}", expectedVersion, result.Output);
+                }
+            }
+        }
+        else
+        {
+            logger.LogInformation("Amazon.Lambda.TestTool version {version} already installed", installedVersion);
+        }
+    }
+
+    internal static bool ShouldInstall(string currentInstalledVersion, string expectedVersionStr, bool allowDowngrading)
+    {
+        if (string.IsNullOrEmpty(currentInstalledVersion))
+        {
+            return true;
+        }
+
+        var installedVersion = Version.Parse(currentInstalledVersion.Replace("-preview", string.Empty));
+        var expectedVersion = Version.Parse(expectedVersionStr.Replace("-preview", string.Empty));
+
+        return (installedVersion < expectedVersion) || (allowDowngrading && installedVersion != expectedVersion);
+    }
+
+    private async Task<string> GetCurrentInstalledVersionAsync(CancellationToken cancellationToken)
+    {
+        var results = await processCommandService.RunProcessAndCaptureOuputAsync(logger, "dotnet", "lambda-test-tool --tool-info", cancellationToken);
```

Comment:
might need to update based on other PR changes https://github.com/aws/aws-lambda-dotnet/pull/1969

Reply:
Done